Playmate/Recommender System/RecommenderSystem.m

```diff
@@ -0,0 +1,94 @@
+//
+//  RecommenderSystem.m
+//  Playmate
+//
+//  Created by Jocelyn Tseng on 8/1/22.
+//
+
+#import "RecommenderSystem.h"
+#import "QuizResult.h"
+#import "Session.h"
+#import "Helpers.h"
+#import "Strings.h"
+
+@implementation RecommenderSystem
+
++ (NSArray *)runRecommendationAlgorithm {
+    
+    PFUser *user = [[PFUser currentUser] fetchIfNeeded];
+    
+    // query every single session
+    __block NSArray *sessions;
+    PFQuery *query = [PFQuery queryWithClassName:@"SportsSession"];
+    [query findObjectsInBackgroundWithBlock:^(NSArray *objects, NSError *error) {
+        if (error != nil) {
+            [Helpers handleAlert:error withTitle:[Strings errorString] withMessage:nil forViewController:self];
+        } else {
+            sessions = objects;
+        }
+    }];
+    
+    NSMutableArray *playSports = [[NSMutableArray alloc] init];
+    
+    NSString *resultId = [user objectForKey:@"quizResult"][0];
+    // If user has completed quiz
+    if (resultId != nil) {
+        QuizResult *result = [[PFQuery getObjectOfClass:@"QuizResult" objectId:resultId] fetchIfNeeded];
+        
+        // filter by location
+        Location *location = result.location;
+        sessions = [RecommenderSystem filterSessions:sessions withLocation:location andRadius:[NSNumber numberWithInt:50]];
+        
+        NSArray *dontPlaySports = result.dontPlaySportsList;
+        NSMutableArray *filteredSessions = [[NSMutableArray alloc] init];
+        
+        // filter out sports the user does not play
+        for (Session *session in sessions) {
+            if (![dontPlaySports containsObject:session.sport]) {
+                [filteredSessions addObject:session];
+            }
+        }
```

Comment:
I'd probably move the quiz filter and the sports don't played to separate functions in case something changes in the future you can directly address changes from there

Reply:
Could you please clarify what you mean by this?